src/vmm/src/guest_config/templates/mod.rs

```diff
@@ -4,26 +4,59 @@
 // TODO: Refactor code to merge deserialize_* functions for modules x86_64 and aarch64
 /// Templates module to contain sub-modules for aarch64 and x86_64 templates
 
+#[cfg(target_arch = "aarch64")]
+mod common_types {
+    use crate::guest_config::aarch64::static_cpu_templates::Aarch64StaticCpuTemplates;
+    use crate::guest_config::aarch64::{Aarch64CpuConfiguration, Error};
+    use crate::guest_config::templates::aarch64::Aarch64CpuTemplate;
+
+    /// Type alias agnostic of CPU architecture for CPU configuration
+    pub type CpuConfiguration = Aarch64CpuConfiguration;
+    /// Type alias agnostic of CPU architecture for custom CPU templates
+    pub type CustomCpuTemplate = Aarch64CpuTemplate;
+    /// Type alias agnostic of CPU architecture for static CPU templates
+    pub type StaticCpuTemplate = Aarch64StaticCpuTemplates;
+
+    /// Type alias agnostic of CPU architecture for Error
+    pub type GuestConfigError = Error;
```

Comment:
Not really. If we move this line out of `common_types` then in `common_types` we would need to have `pub use crate::guest_config::aarch64::Error` (same for x86). So instead of 2 lines now it would be 3.

Reply:
This is only the case if the types are within a module, if moved out of the module this removes these lines. I would tend towards a flatter structure but I understand this would require more instance of `cfg`. I don't have a strong opinion either way here.